program/src/state.rs

```diff
@@ -1524,7 +1524,7 @@ impl MangoAccount {
     pub fn check_open_orders(
         &self,
         mango_group: &MangoGroup,
-        open_orders_ais: &[AccountInfo; MAX_PAIRS],
+        open_orders_ais: &[AccountInfo],
     ) -> MangoResult {
         for i in 0..mango_group.num_oracles {
             if self.in_margin_basket[i] {
```

Comment:
It looks like this would produce a bounds error when accessing `open_orders_ais[i]` for a token index that's in_margin_basket if not all open orders accounts are passed.

Maybe:
- use a separate index `j` for looking into `open_orders_ai`
- skip all `open_orders_ai[j] == Pubkey::default()` (to preserve the old behavior when an account is passed for each pair)
- if `spot_open_orders[i] == Pubkey::default()`: don't advance `j`

To make it possible to support the old case as well as the new case where you send only the open orders accounts that are needed?

Reply:
As discussed this creates a lot of issues with backwards compatibility as check_open_orders  and get_val method is used across a lot of different instructions and there are edge cases like open orders accounts being stored on the mango account even when a token is not in the margin basket.